### PR TITLE
 Added BFS Solution for Building Road Problem in Silver Section

### PR DESCRIPTION
**Pull request by D-ivyanshu**
Todo -> added the bfs code in silver section for the building road (CSES) problem 

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
  

Attaching image for reference

bfs solution needed

### CODE REVIEW

File: content/3_Silver/Graph_Traversal.mdx [hunk 1]
```diff
@@ -619,7 +619,60 @@ See [here.](https://codeforces.com/blog/entry/67883)
...
+}
+
+int main() {
+	scanf("%d%d", &N, &M);
+	for (int i = 0, u, v; i < M; ++i)
+		scanf("%d%d", &u, &v), adj_list[u].push_back(v),
+		    adj_list[v].push_back(u);
+	ans = count_components();
+	printf("%d\n", ans - 1);
+	for (int i = 1; i < ans; ++i) printf("%d %d\n", rep[i - 1], rep[i]);
```
Comment (SansPapyrus683) [inline content/3_Silver/Graph_Traversal.mdx:670]: braces

File: content/3_Silver/Graph_Traversal.mdx [hunk 2]
```diff
@@ -619,7 +619,60 @@ See [here.](https://codeforces.com/blog/entry/67883)
...
+			bfs(i);
+		}
+	return count;
+}
+
+int main() {
+	scanf("%d%d", &N, &M);
+	for (int i = 0, u, v; i < M; ++i)
+		scanf("%d%d", &u, &v), adj_list[u].push_back(v),
+		    adj_list[v].push_back(u);
```
Comment (SansPapyrus683) [inline content/3_Silver/Graph_Traversal.mdx:667]: statements on separate lines

File: content/3_Silver/Graph_Traversal.mdx [hunk 3]
```diff
@@ -619,7 +619,60 @@ See [here.](https://codeforces.com/blog/entry/67883)
...
+	ans = count_components();
+	printf("%d\n", ans - 1);
+	for (int i = 1; i < ans; ++i) printf("%d %d\n", rep[i - 1], rep[i]);
+	return 0;
```
Comment (SansPapyrus683) [inline content/3_Silver/Graph_Traversal.mdx:671]:
```suggestion
```

File: content/3_Silver/Graph_Traversal.mdx (hunk 2 above)
Comment (SansPapyrus683) [reply content/3_Silver/Graph_Traversal.mdx:667]: also braces

File: content/3_Silver/Graph_Traversal.mdx [hunk 4]
```diff
@@ -619,7 +619,60 @@ See [here.](https://codeforces.com/blog/entry/67883)
...
+	for (int i = 1; i <= N; ++i)
+		if (!visited[i]) {
+			rep[count++] = i;
+			bfs(i);
+		}
+	return count;
+}
+
+int main() {
+	scanf("%d%d", &N, &M);
```
Comment (SansPapyrus683) [inline content/3_Silver/Graph_Traversal.mdx:664]: use `cin` for consistency

File: content/3_Silver/Graph_Traversal.mdx [hunk 5]
```diff
@@ -619,7 +619,60 @@ See [here.](https://codeforces.com/blog/entry/67883)
...
+<CPPSection>
+
+```cpp
+#include <cstdio>
+#include <queue>
+#include <vector>
+
+const int MN = 1e5 + 10;
+
+int N, M, ans, rep[MN];
```
Comment (SansPapyrus683) [inline content/3_Silver/Graph_Traversal.mdx:631]: declarations on separate lines

File: content/3_Silver/Graph_Traversal.mdx [hunk 6]
```diff
@@ -619,7 +619,60 @@ See [here.](https://codeforces.com/blog/entry/67883)
...
 
-<IncompleteSection />
+<CPPSection>
+
+```cpp
+#include <cstdio>
+#include <queue>
+#include <vector>
+
+const int MN = 1e5 + 10;
```
Comment (SansPapyrus683) [inline content/3_Silver/Graph_Traversal.mdx:629]: why +10

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: content/3_Silver/Graph_Traversal.mdx (hunk 5 above)
Comment (D-ivyanshu) [reply content/3_Silver/Graph_Traversal.mdx:631]: I have tried to make the bfs solution similar to the dfs one already uploaded on the silver section in graph traversal. You can check that in the silver section (Graph traversal).

File: content/3_Silver/Graph_Traversal.mdx (hunk 4 above)
Comment (D-ivyanshu) [reply content/3_Silver/Graph_Traversal.mdx:664]: but scanf is used in the module in the dfs code.

File: content/3_Silver/Graph_Traversal.mdx (hunk 1 above)
Comment (D-ivyanshu) [reply content/3_Silver/Graph_Traversal.mdx:670]: https://usaco.guide/silver/graph-traversal?lang=cpp (review the dfs code of building roads here)

Comment (SansPapyrus683) [reply content/3_Silver/Graph_Traversal.mdx:670]: i'll change that code later, here add braces tho

File: content/3_Silver/Graph_Traversal.mdx (hunk 5 above)
Comment (SansPapyrus683) [reply content/3_Silver/Graph_Traversal.mdx:631]: > i'll change that code later

that code itself was kinda bad

File: content/3_Silver/Graph_Traversal.mdx (hunk 4 above)
Comment (SansPapyrus683) [reply content/3_Silver/Graph_Traversal.mdx:664]: but `cin` is used literally everywhere else

File: content/3_Silver/Graph_Traversal.mdx (hunk 5 above)
Comment (D-ivyanshu) [reply content/3_Silver/Graph_Traversal.mdx:631]: ok then I will change the bfs code.

File: content/3_Silver/Graph_Traversal.mdx (hunk 1 above)
Comment (D-ivyanshu) [reply content/3_Silver/Graph_Traversal.mdx:670]: ok

Comment (D-ivyanshu) [reply content/3_Silver/Graph_Traversal.mdx]: @SansPapyrus683  I have made the changes that you have pointed.

**Review by SansPapyrus683 — APPROVED** [review]
good enough, i suppose

**Comment by D-ivyanshu** [comment]
@SansPapyrus683  Is it likely that this pull request will be merged?

**Comment by SansPapyrus683** [comment]
yeah uh- @ryanchou-dev review

**Comment by ryanchou-dev** [comment]
eek ok, we can prolly merge the two prs. kevin already approved this one, so why not :/


at an event right now, so ill merge in two days?

**Comment by D-ivyanshu** [comment]
> 

Thanks ; )

**Comment by ryanchou-dev** [comment]
transferred to #3586 :D

**Pull request closed without merging**